tactics/class_tactics.ml

```diff
@@ -78,6 +78,16 @@ let get_typeclasses_verbose () =
 let set_typeclasses_depth d = (:=) typeclasses_depth d
 let get_typeclasses_depth () = !typeclasses_depth
 
+let typeclasses_caching = ref true
+let set_typeclasses_caching d = (:=) typeclasses_caching d
+let get_typeclasses_caching () = !typeclasses_caching
+
+(** [typeclasses_caching_min_subgoals] controls how much subgoals had to be checked before failed goal be cached. It is performance optimzation to avoid caching 'easy' goals, where overhad of the cache overcomes it's potential benefits. *)
```

Comment:
Split comments onto separate lines (avoid lines longer than 80 characters).
Typos:
- how much -> how many
- had to -> have to
- be cached -> is cached
- overhad -> overhead
- it's potential benefits -> its potential benefits